pkg/traceability/sampling/sample.go

```diff
@@ -20,9 +21,28 @@ func (s *sample) ShouldSampleTransaction(details TransactionDetails) bool {
 	if hasFailedStatus && s.config.ReportAllErrors {
 		return true
 	}
-	if s.config.PerAPI && details.APIID != "" {
-		return s.shouldSampleWithCounter(details.APIID)
+
+	if s.config.PerAPI && s.config.PerSub {
+		if details.SubID != "" {
+			return s.shouldSampleWithCounter(fmt.Sprintf("%s-%s", details.APIID, details.SubID))
```

Comment:
@jcollins-axway  like this?

Reply:
that looks better, think you can drop the conditional on 29 and just return then

Reply:
that looks better, think you can drop the conditional on 29 and just return then

Reply:
We still need to check if there is an api id though?

Reply:
true true, that works